src/api/app/models/obs_factory/distribution_strategy_opensuse_leap15.rb

```diff
@@ -43,13 +43,10 @@ def totest_version_file
     #
     # @return [String] version string
     def published_version
-      begin
-        f = URI(repo_url).open
-      rescue ::OpenURI::HTTPError
-        return 'unknown'
-      end
-      matchdata = %r{openSUSE-#{opensuse_leap_version}-#{published_arch}-Build(.*)-Media}.match(f.read)
-      matchdata[1]
+      response = Net::HTTP.get_response(URI.parse(repo_url))
+      return 'unknown' if response.code !== Net::HTTPSuccess
```

Comment:
as Leap15 inherits the FactoryStrategy, I would recommend to move these 2 lines in a function (DRY)